src/pages/Admin/Charity/Permissions/Table.tsx

```diff
@@ -0,0 +1,129 @@
+  profile: {} as FormField,
+};
+const FORM_KEYS = getTypedKeys(formValues);
+
+export default function Table({ className = "", disabled = false }) {
+  const { endow_type } = useAdminResources<"charity">();
+  const { wallet } = useGetWallet();
+
+  const {
+    register,
+    watch,
+    formState: { errors },
+  } = useFormContext<FormValues>();
+
+  const isNormal = endow_type === "normal";
+
+  return (
+    <table
+      className={`${className} w-full text-sm rounded border border-separate border-spacing-0 border-prim`}
+    >
+      <TableSection
+        type="thead"
+        rowClass="bg-orange-l6 dark:bg-blue-d7 divide-x divide-prim"
+      >
+        <Cells
+          type="th"
+          cellClass="py-3 px-4 text-xs uppercase font-semibold text-left first:rounded-tl last:rounded-tr"
+        >
+          <>Action</>
+          <>Admin wallet</>
+          {isNormal ? <>Governance</> : null}
+          <>Delegate</>
+          <>Delegate address</>
+          <>Actions</>
+        </Cells>
+      </TableSection>
+      <TableSection
+        type="tbody"
+        rowClass="dark:bg-blue-d6 divide-x divide-prim"
+        selectedClass="bg-orange-l5 dark:bg-blue-d4"
+      >
+        {FORM_KEYS.map((fieldName) => {
+          const delegate = watch(`${fieldName}.delegate`);
+          const name = watch(`${fieldName}.name`);
+          const initDelegate = watch(`initialValues.${fieldName}.delegate`);
+          const initDelegateAddress = watch(
+            `initialValues.${fieldName}.delegate_address`
+          );
+          const initModifiable = watch(`initialValues.${fieldName}.modifiable`);
+          const modifiable = watch(`${fieldName}.modifiable`);
+
+          const isDisabled =
+            disabled ||
+            !initModifiable ||
+            (initDelegate &&
+              !!initDelegateAddress &&
+              initDelegateAddress !== wallet?.address);
+
+          const inputDisabled = isDisabled || !modifiable;
+
+          return (
+            <Cells
+              key={`table-row-${fieldName}`}
+              type="td"
+              cellClass="py-3 px-4 border-t border-prim min-w-[116px] max-w-xs truncate first:rounded-bl last:rounded-br"
+            >
+              <>{name}</>
+
+              <CheckField<FormValues>
+                name={`${fieldName}.owner_controlled`}
+                classes={{
+                  label: "uppercase text-xs font-bold",
+                  input: "checkbox-orange",
+                }}
+                disabled={inputDisabled}
+              />
+              {isNormal ? (
+                <CheckField<FormValues>
+                  name={`${fieldName}.gov_controlled`}
+                  classes={{
+                    label: "uppercase text-xs font-bold",
+                    input: "checkbox-orange",
+                  }}
+                  disabled={inputDisabled}
+                />
+              ) : null}
+              <CheckField<FormValues>
```

Comment:
ISSUE: 
having previously set a delegate, 
<img width="1160" alt="image" src="https://user-images.githubusercontent.com/89639563/225830653-a4c615c8-1ba7-41b9-b15b-37412c841e30.png">

clicking `delegate` checkbox should clear `input` before disabling

REPRO: go to `/admin/27/permissions`

Reply:
@SovereignAndrey @ap-justin we can leave the value in the input just on UI side, but not submit it with the rest of the data if `delegate` was set to `false` - this way the user is able to reenable the `delegate` checkbox without losing the address value?